api/rpc/account/account.go

```diff
@@ -0,0 +1,357 @@
+	}
+	r.Email = address.Address
+	return
+}
+
+// Server is used to implement account.AccountServer
+type Server struct{}
+
+// SignUp implements account.SignUp
+func (s *Server) SignUp(ctx context.Context, in *SignUpRequest) (out *SessionReply, err error) {
+	out = &SessionReply{}
+	err = in.Validate()
+	if err != nil {
+		return nil, err
+	}
+	_, err = passlib.Hash(in.Password)
+	if err != nil {
+		return nil, grpc.Errorf(codes.Internal, "hashing error")
+	}
+	out.SessionKey = in.Name
+	return
+}
+
+// Verify implements account.Verify
+func (s *Server) Verify(ctx context.Context, in *VerificationRequest) (out *pb.Empty, err error) {
+	out = &pb.Empty{}
+	err = in.Validate()
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println(in.Code)
+	return
+}
+
+// CreateOrganization implements account.CreateOranization
+func (s *Server) CreateOrganization(ctx context.Context, in *OrganizationRequest) (out *pb.Empty, err error) {
+	out = &pb.Empty{}
+	err = in.Validate()
+	if err != nil {
+		return nil, err
+	}
+	return
+}
+
+// Login implements account.Login
+func (s *Server) Login(ctx context.Context, in *LogInRequest) (out *SessionReply, err error) {
+	out = &SessionReply{}
+	if in.Name == "" {
+		return nil, grpc.Errorf(codes.InvalidArgument, "organization name is mandatory")
+	}
+	_, err = passlib.Verify(in.Password, hash)
+	if err != nil {
+		return nil, grpc.Errorf(codes.Unauthenticated, err.Error())
+	}
+	out.SessionKey = in.Name
+	return
+
+}
+
+// Switch implements account.Switch
+func (s *Server) Switch(ctx context.Context, in *TeamRequest) (out *pb.Empty, err error) {
+	out = &pb.Empty{}
+	if in.Organization == "" {
+		return nil, grpc.Errorf(codes.InvalidArgument, "organization name is mandatory")
+	}
+	return
+}
+
+// ListAccounts implements account.ListAccounts
+func (s *Server) ListAccounts(ctx context.Context, in *AccountsRequest) (out *ListReply, err error) {
+	out = &ListReply{}
+	if in.Type != "individual" && in.Type != "organization" {
+		return nil, grpc.Errorf(codes.InvalidArgument, "account type is mandatory")
+	}
+	return
+}
+
+// GetAccountDetails implements account.GetAccountDetails
+func (s *Server) GetAccountDetails(ctx context.Context, in *AccountRequest) (out *AccountReply, err error) {
+	out = &AccountReply{}
+	if in.Name == "" {
+		return nil, grpc.Errorf(codes.InvalidArgument, "name is mandatory")
+	}
+	return
+}
+
+// EditAccount implements account.EditAccount
+func (s *Server) EditAccount(ctx context.Context, in *EditRequest) (out *AccountReply, err error) {
+	out = &AccountReply{}
+	if in.Name == "" {
+		return nil, grpc.Errorf(codes.InvalidArgument, "name is mandatory")
+	}
+	if in.Email != "" {
+		_, err := mail.ParseAddress(in.Email)
+		if err != nil {
+			return nil, grpc.Errorf(codes.InvalidArgument, err.Error())
+		}
+	}
+	if in.NewPassword != "" {
+		_, err := passlib.Verify(in.Password, hash)
+		if err != nil {
+			return nil, grpc.Errorf(codes.Unauthenticated, err.Error())
+		}
+		if len(in.NewPassword) < 8 {
```

Comment:
These kinds of things should be in the validators. Validators can be parameterized if necessary to limit validation scope depending on operation. Validators can chain calls to fundamental Validate functions, like ValidatePassword, or perhaps validators types, like PasswordValidator with a Validate method. Not trying to be too hard on this PR, but since this will likely be the model for all the resources to follow, we probably want to get it right.